navigation/plot.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+// Standard C++ includes
+#include <algorithm>
+#include <vector>
+
+// Third-party includes
+#include "third_party/matplotlibcpp.h"
+
+namespace BoBRobotics {
+namespace Navigation {
+
+//! Plot a line graph of a single RIDF
+void
+plotRIDF(const std::vector<float> &differences)
+{
+    // Vector to store modified RIDF
+    std::vector<float> ridf(differences.size() + 1);
+
+    // Rotate so data is in range -180 to 180
+    std::rotate_copy(differences.begin(),
```

Comment:
You don't necessarily need to _resize_ ``ridf`` just _reserve_ it (i.e. you don't care what it's initial values are) so you could do:
```c++
std::vector<float> ridf;
ridf.reserve(differences.size() + 1);
std::rotate_copy(differences.begin(),
    differences.begin() + ceil((float) differences.size() / 2.0f),
    differences.end(),
    std::back_inserter(ridf));
```

Reply:
Good spot!